services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -51,11 +51,21 @@ export const useApplyActions = (): Promise<Empty> => useApi.batchService().Proce
 
 export const useActions = (): BatchAction[] => useAction(({ actions }) => actions);
 
+export const [useSidebar, UpdateSidebar] = createStore({ shown: false });
+
+export const useSidebarShown = (): boolean => useSidebar(({ shown }) => shown);
+
+export const useNumberOfActions = (): number => useAction(({ actions }) => actions.length);
+
 export const updateActions = (actions: BatchAction[]): void => {
     deleteAllActions();
     actions.forEach((action) => addAction(action));
 };
 
+export const appendAction = (actions: BatchAction[]): void => {
```

Comment:
What does this function do?

Reply:
It appends to the actions list, unlike updateActions which deletes all actions and adds another actions.
It is used in the unit testing of the sidebar.

Reply:
Is there any difference between `updateActions` and `appendActions` where your code is concerned? 
in other words, what are you trying to test? 

I think if you mock everything it would make your unit tests simpler.

Reply:
I am testing the functionality of showing the number of actions in the cart. I first add 2 actions then make sure that the reading is 2, then add another action and check that it is 3, and so on. This isn't achievable with updateActions as it clears all actions first which would always reset the number of actions to before adding them.

Reply:
adding an action basically just changes the array. 
the function of automatic refresh and (update the number from 2 to 3) is already included in the store library. there's no need to test this. 

what you need to test is what your code does. 
e.g. if there's no actions title is 'Planned Actions'
if there's two then it's 'Planned Actions (2)'
etc. 

the hook simply uses the array.length which is just part of the language. also not needed to test. 

check [this](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/WORKSPACE_DOCUMENT/-NIHiZUxqRbTHNflVb69/) out I hope it helps.